File: recetas-backend/src/main/java/com/silvia/recetas/model/Usuario.java
package com.silvia.recetas.model;

import jakarta.persistence.*;
import lombok.Data;
import java.util.List;

@Entity
@Table(name = "usuarios")
@Data
public class Usuario {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(unique = true, nullable = false)
    private String username;

    @Column(nullable = false)
    private String password;

    @Column(nullable = false)
    private String nombre;

    @OneToMany(mappedBy = "usuario", cascade = CascadeType.ALL)
    private List<Receta> recetas;
}

File: recetas-frontend/src/app/models/usuario.ts
export interface Usuario {
    id?: number;
    username: string;
    nombre: string;
    password?: string;
}

export interface LoginRequest {
    username: string;
    password: string;
}

export interface LoginResponse {
    token: string;
    username: string;
    nombre: string;
}

export interface RegisterRequest {
    username: string;
    nombre: string;
    password: string;
    confirmPassword?: string;
}